feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/reply/ReplyTextFieldVisibilityUITest.kt

```diff
@@ -0,0 +1,168 @@
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@Category(EndToEndTest::class)
+@RunWith(RobolectricTestRunner::class)
+class ReplyTextFieldVisibilityUITest : BaseComposeTest() {
+
+    private val clientId = 20L
+
+    private val course: Course = Course(id = 1)
+    private val conversation = OneToOneChat(id = 2)
+
+    private val answers = (0..2).map { index ->
+        AnswerPostPojo(
+            parentPostId = "client-id",
+            postId = "answer-client-id-$index",
+            resolvesPost = false,
+            basePostingCache = AnswerPostPojo.BasePostingCache(
+                serverPostId = index.toLong(),
+                authorId = clientId,
+                creationDate = Clock.System.now(),
+                updatedDate = null,
+                content = "Answer Post content $index",
+                authorRole = UserRole.USER,
+                authorName = "author name"
+            ),
+            reactions = emptyList(),
+            serverPostIdCache = AnswerPostPojo.ServerPostIdCache(
+                serverPostId = index.toLong()
+            )
+        )
+    }
+
+    private val posts = (0..2).map { index ->
+        PostPojo(
+            clientPostId = "client-id-$index",
+            serverPostId = index.toLong(),
+            content = "Post content $index",
+            resolved = false,
+            updatedDate = null,
+            creationDate = Clock.System.now(),
+            authorId = clientId,
+            title = null,
+            authorName = "author name",
+            authorRole = UserRole.USER,
+            courseWideContext = null,
+            tags = emptyList(),
+            answers = if (index == 0) answers else emptyList(),
+            reactions = emptyList()
+        )
+    }
+
+    @Test
+    fun `test GIVEN the thread view is shown containing one post and three answer posts WHEN the markdown text field is clicked THEN the keyboard is shown below the markdown text field`() {
+        setupThreadUi(posts[0])
+        runTest()
+    }
+
+    @Test
+    fun `test GIVEN the chat list containing three posts is shown WHEN the markdown text field is clicked THEN the keyboard is shown below the markdown text field`() {
+        setupChatUi(posts)
+        runTest()
+    }
+
+    private fun runTest() {
+        val markdownTextField = composeTestRule.onNodeWithTag(TEST_TAG_CAN_CREATE_REPLY)
+        val initialPosition = markdownTextField.fetchSemanticsNode().positionInRoot.y
+
+        markdownTextField.performClick()
+        composeTestRule.waitForIdle()
+
+        val newPosition = markdownTextField.fetchSemanticsNode().positionInRoot.y
+
+        markdownTextField
+            .assertExists()
+            .assertIsDisplayed()
+        assertTrue("Text field should move up when the keyboard appears", newPosition < initialPosition)
+    }
+
+    private fun setupThreadUi(post: PostPojo) {
```

Comment:
A very similar method exists in `ConversationAnswerMessagesUITest` maybe create common superclass (could also contain the "setup" of the post and answer pojos)